src/payment/strategies/googlepay/googlepay-payment-strategy.ts

```diff
@@ -134,81 +145,50 @@ export default class GooglePayPaymentStrategy implements PaymentStrategy {
-        }
+        let state = this._store.getState();
+        this._paymentMethod = state.paymentMethods.getPaymentMethodOrThrow(methodId);
 
         let nonce;
 
-        if (this._methodId === 'googlepayadyenv2') {
+        if (methodId === 'googlepayadyenv2') {
             nonce = JSON.stringify({
                 type: AdyenPaymentMethodType.GooglePay,
-                googlePayToken: paymentMethod.initializationData.nonce,
+                googlePayToken: this._paymentMethod.initializationData.nonce,
                 browser_info: getBrowserInfo(),
             });
         } else {
-            nonce = paymentMethod.initializationData.nonce;
+            nonce = this._paymentMethod.initializationData.nonce;
         }
-
         const paymentData = {
-            method: this._methodId,
+            method: methodId,
             nonce,
-            cardInformation: paymentMethod.initializationData.card_information,
+            cardInformation: this._paymentMethod.initializationData.card_information,
         };
 
+        if (this._paymentMethod.initializationData.nonce) {
+            state = await this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(methodId));
+            this._paymentMethod = state.paymentMethods.getPaymentMethodOrThrow(methodId);
+        }
+
         return {
-            methodId: this._methodId,
+            methodId,
             paymentData,
         };
     }
 
-    @bind
-    private _handleWalletButtonClick(event: Event): Promise<InternalCheckoutSelectors> {
-        event.preventDefault();
-
-        if (!this._methodId || !this._googlePayOptions) {
-            throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
-        }
-
-        const {
-            onError = () => {},
-            onPaymentSelect = () => {},
-        } = this._googlePayOptions;
-
-        return this._store.dispatch(this._paymentStrategyActionCreator.widgetInteraction(() => {
-            return this._googlePayPaymentProcessor.displayWallet()
-                .then(paymentData => this._paymentInstrumentSelected(paymentData))
-                .then(() => onPaymentSelect())
-                .catch(error => {
-                    if (error.statusCode !== 'CANCELED') {
-                        onError(error);
-                    }
-                });
-        }, { methodId: this._methodId }), { queueId: 'widgetInteraction' });
-    }
-
-    private async _paymentInstrumentSelected(paymentData: GooglePaymentData) {
-        if (!this._methodId) {
+    private async _paymentInstrumentSelected(paymentData: GooglePaymentData, methodId: string) {
+        if (!methodId) {
             throw new NotInitializedError(NotInitializedErrorType.PaymentNotInitialized);
         }
 
-        const methodId = this._methodId;
-
         // TODO: Revisit how we deal with GooglePaymentData after receiving it from Google
         await this._googlePayPaymentProcessor.handleSuccess(paymentData);
 
```

Comment:
Remove this line break

Reply:
it's needed before a `return` statement (eslint rules)